<table align="left">
  <td>
    <a href="https://colab.research.google.com/github/marco-canas/calculo/blob/main/classes/3_calculo_for_eng_licenciatura_administracion/4_integration/1_antidiferenciacion/2_aplicaciones_de_la_antidiferenciacion/1_caida_libre/2_evaluacion_oral_tarea_quiz_parcial_sobre_caida_libre.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>
  </td>
</table>

### [Video de apoyo a la lectura interactiva y experimental de este cuaderno](https://www.youtube.com/watch?v=qd5LOAjCwFw)

## [Vínculo al programa del curso:](https://github.com/marco-canas/calculo/blob/main/2_programa_calculo_Integral_ingenieria_agropecuaria.docx)

# Modelo general de clase: [aula invertida](https://www3.gobiernodecanarias.org/medusa/ecoescuela/pedagotic/aprendizaje-invertido-flipped-classroom/)


# Fomento de la lectura literaria asociada a Ingeniería agropecuaria y licenciatura en Matemáticas  

[Hábitos atómicos](https://github.com/marco-canas/habitos_atomicos/blob/main/1_el_sorprendente_poder_de_los_habitos_atomicos/1_el_sorprendente_poder_de_los_habitos_atomicos.ipynb)

# Título de la clase: Realización evaluación oral, de Tareas en cuadernos Jupyter, Quiz o Parcial de seleccipon multiple.  



### Objetivo de aprendizaje (Enunciar, de manera corta sencilla y clara para todos los asistentes)

* Evaluar de manera integral y continua, el aprendizaje de los estudiantes en cuanto al concepto de antiderivada y su aplicación al fenómeno de caída de los cuerpos
* Evaluar la capacidad de describir fenómenos físicos utilizando funciones de velocidad y altura. 

## Capacidades a evaluar el día de hoy:

1. Capacidad de resolver ejercicios de aplicación de la antiderivada a la física.
2. Capacidad de justificar los procedimientos que realiza de manera escrita. 
3. Capacidad para describir el proceso de solución de manera oral, de tal manera que se haga entender. 
4. Capacidad de representar funciones de manera algebriaca, tabular y gráfica.
5. Capacidad de interpretar cada una de estas representaciones. 

## Establecimiento de prerrequisitos conceptuales para alcanzar el objetivo de aprendizaje trazado.   

(Elabore una lista de preguntas al rededor de los conceptos que consideres necesarios para empezar la clase).  

Aquí tienes una lista de prerrequisitos conceptuales para alcanzar el objetivo de aplicar el concepto de integración indefinida al fenómeno físico de caída de los cuerpos utilizando cuadernos Jupyter, lenguaje LaTeX y lenguaje Python:



1. **Comprensión básica de la física del movimiento:** Es fundamental tener una comprensión básica de los conceptos físicos relacionados con el movimiento de los cuerpos, como la posición, velocidad y aceleración, así como las leyes del movimiento de Newton.



2. **Conocimientos de cálculo diferencial:** Para comprender la integración indefinida y su aplicación en el contexto de la física, es necesario tener una comprensión sólida de los conceptos de cálculo diferencial, como la derivada y la tasa de cambio instantánea.



3. **Familiaridad con la integración indefinida:** Debes estar familiarizado con el concepto de integral indefinida y su relación con la antiderivada. Es importante comprender cómo calcular integrales indefinidas y cómo interpretar el resultado en el contexto del problema físico.



4. **Habilidades en el uso de cuadernos Jupyter:** Debes estar cómodo utilizando cuadernos Jupyter para escribir y ejecutar código Python, así como para incorporar texto en formato LaTeX para explicar conceptos matemáticos y físicos de manera clara y concisa.



5. **Conocimientos básicos de programación en Python:** Es esencial tener conocimientos básicos de programación en Python, incluyendo la sintaxis del lenguaje, estructuras de control de flujo, funciones y manipulación de datos, ya que utilizarás Python para realizar cálculos numéricos y graficar resultados.

6. **Capacidad para modelar fenómenos físicos:** Debes ser capaz de utilizar conceptos matemáticos y físicos para modelar fenómenos físicos en forma de ecuaciones y expresiones matemáticas, lo que te permitirá aplicar la integración indefinida de manera efectiva en el análisis de estos fenómenos.

7. **Conocimientos básicos de LaTeX:** Es útil tener conocimientos básicos de LaTeX para formatear correctamente el texto y las ecuaciones matemáticas en los cuadernos Jupyter, lo que facilitará la comunicación clara de ideas y resultados.

8. **Habilidades de análisis y resolución de problemas:** Debes ser capaz de analizar problemas físicos complejos y aplicar técnicas de resolución de problemas para desarrollar modelos matemáticos y encontrar soluciones utilizando la integración indefinida y Python.

Al tener un sólido entendimiento de estos prerrequisitos conceptuales, estarás bien preparado para aplicar el concepto de integración indefinida al fenómeno físico de caída de los cuerpos utilizando cuadernos Jupyter, LaTeX y Python.

# Llamada aleatoria para los estudiantes a participar para ser evaluados de manera oral



In [1]:
import pandas as pd
import random

# Cargar la lista de estudiantes desde la URL
url = 'https://raw.githubusercontent.com/marco-canas/groups_list/main/calculo_integral.csv'
estudiantes = pd.read_csv(url)

# Función para llamar a un estudiante aleatoriamente y registrar su nota de seguimiento
def llamar_estudiante():
    estudiante_aleatorio = random.choice(estudiantes['Nombre'])
    print(f"Llamando a {estudiante_aleatorio}...")
    
    
# Llamar a la función para realizar el proceso
llamar_estudiante()

Llamando a PEREZ THERAN ERLINDA...


## Preguntas en secuencia para la evaluación oral



# Para el diseño de la clase tenga en cuenta algunos problemas de aprendizaje

* El estudoiante partde de fótmulas de cinemética, sin deducirlas en este curso de cálculo integral. 
* 

# Tarea asignada o taller para evaluar

## Tareas para realizar antes de clase



In [6]:
## Tareas para realizar antes de clase

import pandas as pd
import numpy as np
np.random.seed(1213)

texto_guia_seccion_pagina = 'Purcell_seccion_5.6_pagina 313'
n_exercises_del_taller_del_texto = 5
n_ejercicios_asignados_al_estudiante = 1

# Descargar la lista de estudiantes y ejercicios
students_url = "https://raw.githubusercontent.com/marco-canas/groups_list/main/calculo_integral.csv"
exercises = list(range(1, n_exercises_del_taller_del_texto + 1))  # Lista de ejercicios del 1 al 83

# Cargar la lista de estudiantes
students_data = pd.read_csv(students_url)

# Crear una lista para almacenar las asignaciones
assignments = []

# Asignar ejercicios aleatorios a cada estudiante
for index, student in students_data.iterrows():
    random_exercises = np.random.choice(exercises, n_ejercicios_asignados_al_estudiante, \
                                        replace=False)  # Seleccionar 5 ejercicios aleatorios sin repetición
    assignments.append([students_data["Nombre"][index], *random_exercises])  # Agregar asignación a la lista

# Crear un DataFrame con las asignaciones
columns = ["Student", "Example"]
assignment_df = pd.DataFrame(assignments, columns=columns)

# Guardar el DataFrame en un archivo CSV
assignment_csv = texto_guia_seccion_pagina 
assignment_df.to_csv(assignment_csv, index=False)

print("Asignaciones guardadas en", assignment_csv)
assignment_df


Asignaciones guardadas en Purcell_seccion_5.6_pagina 313


Student  Example
0        AGUDELO MORENO RENATO DIRNEY        3
1   AMARIS FERNANDEZ SEBASTIAN CAMILO        2
2       ARENAS HERAZO GABRIEL ENRIQUE        2
3        AYALA MENDOZA MARISOL YACIRA        1
4          BERTEL RODELO MICHELL SMIT        5
5            BULA VERGARA MARIA PAULA        5
6             CANTILLO ENSUCHO DANIEL        1
7            CERVANTES DURAN ANGELICA        5
8         GALLEGO TEJADA JUAN ESTEBAN        1
9             GUERRA NORIEGA YONATHAN        3
10    HINCAPIE CASTRO JOHAN SEBASTIAN        3
11             HUMANEZ SANCHEZ CAMILO        5
12       IBAnEZ MORENO FRANCISCO LUIS        3
13  LONDONO CUADRADO CRISTIAN ALBERTO        1
14       LOZANO HOLGUIN LIZETH JOHANA        3
15          MASS MIRANDA RIVALDO JOSE        4
16     MIRANDA MARTINEZ CARLOS ANDRES        1
17              MONTERROSA SOTO ABRIL        3
18           NORIEGA JIMENEZ GABRIELA        1
19             PEREZ REYES IVAN DARIO        3
20               PEREZ THERAN ERLINDA        2
21              PINEDA ROMERO YOSMERI        4
22         QUINONEZ CERVANTES MARIANA        3
23           QUIROZ DURAN JUAN MANUEL        5
24                 RAMOS PEREZ YANINE        1
25              RIVAS HOYOS ESTEFANIA        4
26              ROSILLO TEJADA VANESA        3
27          SANCHEZ PADILLA SILVESTRE        2
28        SILVA ROYERO MAILY CRISTINA        2
29          USUGA GEORGE JHON ESTIVEN        3

In [15]:
# Tarea en clase

## Quiz de 15 minutos para ejercitación y adquisición de habilidades computacionales y de pensamiento variacional

# Resolver con sympy el punto sobre centro de masa que se le haya propuesto. 

# Puede utilizar chatGPT, pero debe revisar y corregir, lo que este modelo le entregue. 


import pandas as pd
import numpy as np
import random
import requests


texto_guia_seccion_pagina = 'purcell_seccion_5.6_pagina 313'
n_exercises_del_taller_del_texto = 37
n_ejercicios_asignados_al_estudiante = 1



# Descargar la lista de estudiantes y ejercicios
students_url = "https://raw.githubusercontent.com/marco-canas/groups_list/main/calculo_integral.csv"
exercises = list(range(1, n_exercises_del_taller_del_texto))  # Lista de ejercicios del 1 al 83

# Cargar la lista de estudiantes
students_data = pd.read_csv(students_url)

# Crear una lista para almacenar las asignaciones
assignments = []

# Asignar ejercicios aleatorios a cada estudiante
for index, student in students_data.iterrows():
    random_exercises = random.sample(exercises, n_ejercicios_asignados_al_estudiante)  # Seleccionar 5 ejercicios aleatorios sin repetición
    assignments.append([students_data["Nombre"][index], *random_exercises])  # Agregar asignación a la lista

# Crear un DataFrame con las asignaciones
columns = ["Student", "Exer1"]
assignment_df = pd.DataFrame(assignments, columns=columns)

# Guardar el DataFrame en un archivo CSV
assignment_csv = texto_guia_seccion_pagina 
assignment_df.to_csv(assignment_csv, index=False)

print("Asignaciones guardadas en", assignment_csv)
assignment_df

Asignaciones guardadas en purcell_seccion_5.6_pagina 313


Student  Exer1
0        AGUDELO MORENO RENATO DIRNEY      3
1   AMARIS FERNANDEZ SEBASTIAN CAMILO     10
2       ARENAS HERAZO GABRIEL ENRIQUE     34
3        AYALA MENDOZA MARISOL YACIRA     30
4          BERTEL RODELO MICHELL SMIT     31
5            BULA VERGARA MARIA PAULA     23
6             CANTILLO ENSUCHO DANIEL     35
7            CERVANTES DURAN ANGELICA     13
8         GALLEGO TEJADA JUAN ESTEBAN     11
9             GUERRA NORIEGA YONATHAN      1
10    HINCAPIE CASTRO JOHAN SEBASTIAN     32
11             HUMANEZ SANCHEZ CAMILO      7
12       IBAnEZ MORENO FRANCISCO LUIS     20
13  LONDONO CUADRADO CRISTIAN ALBERTO     28
14       LOZANO HOLGUIN LIZETH JOHANA      1
15          MASS MIRANDA RIVALDO JOSE     34
16     MIRANDA MARTINEZ CARLOS ANDRES     12
17              MONTERROSA SOTO ABRIL     30
18           NORIEGA JIMENEZ GABRIELA      1
19             PEREZ REYES IVAN DARIO      7
20               PEREZ THERAN ERLINDA     24
21              PINEDA ROMERO YOSMERI      9
22         QUINONEZ CERVANTES MARIANA     15
23           QUIROZ DURAN JUAN MANUEL     33
24                 RAMOS PEREZ YANINE     22
25              RIVAS HOYOS ESTEFANIA     21
26              ROSILLO TEJADA VANESA     13
27          SANCHEZ PADILLA SILVESTRE     13
28        SILVA ROYERO MAILY CRISTINA      7
29          USUGA GEORGE JHON ESTIVEN     15

In [18]:
## Tarea para despues de clase n de habilidades computacionales y de pensamiento variacional

# Resolver con sympy el punto sobre centro de masa que se le haya propuesto. 

# Puede utilizar chatGPT, pero debe revisar y corregir, lo que este modelo le entregue. 


import pandas as pd
import numpy as np
import random
import requests


texto_guia_seccion_pagina = 'purcell_seccion_3.9_pagina_208'
n_exercises_del_taller_del_texto = 37
n_ejercicios_asignados_al_estudiante = 3



# Descargar la lista de estudiantes y ejercicios
students_url = "https://raw.githubusercontent.com/marco-canas/groups_list/main/calculo_integral.csv"
exercises = list(range(1, n_exercises_del_taller_del_texto))  # Lista de ejercicios del 1 al 83

# Cargar la lista de estudiantes
students_data = pd.read_csv(students_url)

# Crear una lista para almacenar las asignaciones
assignments = []

# Asignar ejercicios aleatorios a cada estudiante
for index, student in students_data.iterrows():
    random_exercises = random.sample(exercises, n_ejercicios_asignados_al_estudiante)  # Seleccionar 5 ejercicios aleatorios sin repetición
    assignments.append([students_data["Nombre"][index], *random_exercises])  # Agregar asignación a la lista

# Crear un DataFrame con las asignaciones
columns = ["Student", "Exer1", "Exer2", "Exer3"]
assignment_df = pd.DataFrame(assignments, columns=columns)

# Guardar el DataFrame en un archivo CSV
assignment_csv = texto_guia_seccion_pagina 
assignment_df.to_csv(assignment_csv, index=False)

print("Asignaciones guardadas en", assignment_csv)
assignment_df

Asignaciones guardadas en purcell_seccion_3.9_pagina_208


Student  Exer1  Exer2  Exer3
0        AGUDELO MORENO RENATO DIRNEY     13      6     36
1   AMARIS FERNANDEZ SEBASTIAN CAMILO     29     36      7
2       ARENAS HERAZO GABRIEL ENRIQUE      3     25     18
3        AYALA MENDOZA MARISOL YACIRA     34     23      9
4          BERTEL RODELO MICHELL SMIT     24      4      6
5            BULA VERGARA MARIA PAULA     35     15     33
6             CANTILLO ENSUCHO DANIEL     13      4      2
7            CERVANTES DURAN ANGELICA      6     13      7
8         GALLEGO TEJADA JUAN ESTEBAN      3     31      9
9             GUERRA NORIEGA YONATHAN     13      2      4
10    HINCAPIE CASTRO JOHAN SEBASTIAN     21      3     19
11             HUMANEZ SANCHEZ CAMILO     20     10     25
12       IBAnEZ MORENO FRANCISCO LUIS     30     33     34
13  LONDONO CUADRADO CRISTIAN ALBERTO     33     35     19
14       LOZANO HOLGUIN LIZETH JOHANA     36     19      8
15          MASS MIRANDA RIVALDO JOSE     20      2     35
16     MIRANDA MARTINEZ CARLOS ANDRES      4     18      1
17              MONTERROSA SOTO ABRIL     13     15      7
18           NORIEGA JIMENEZ GABRIELA     26     12     31
19             PEREZ REYES IVAN DARIO     31      5     19
20               PEREZ THERAN ERLINDA     10     25     35
21              PINEDA ROMERO YOSMERI      9     28     24
22         QUINONEZ CERVANTES MARIANA     25     15     28
23           QUIROZ DURAN JUAN MANUEL      9     22     15
24                 RAMOS PEREZ YANINE      1     23      9
25              RIVAS HOYOS ESTEFANIA     11     21     33
26              ROSILLO TEJADA VANESA      2      9      3
27          SANCHEZ PADILLA SILVESTRE     32     23     21
28        SILVA ROYERO MAILY CRISTINA     15      1     19
29          USUGA GEORGE JHON ESTIVEN     11     30     22

# Ejemplo de quiz

El profesor deja caer un marcador de tablero desde una altura de 2 metros. 

Determine: 
1. Determine la función $v(t)$
2. Determine la función $h(t)$
3. Describa el movimineto de caída utilizando estas dos funciones. 
4. o **prediga** el tiempo que tarda en llegar al suelo. 
5. Y luego, con el tiempo anterior determine o prediga la velocidad con la que llega al suelo. 


In [2]:
# solución 

from sympy import *  # importo la librería de cálculo simbólico

t,v,h, C1 = var('t,v,h, C1') # defino las variables del problema

v = Function('v')(t) # Indica que v es una variable que depende de t

h = Function('h')(t) # indica que la altura h depende del tiempo t.


derivada_de_la_velocidad = -9.8

ecuacion_diferencial = Eq(derivada_de_la_velocidad-diff(v), 0)

ecuacion_diferencial # gracias a Galileo y Newton

Eq(-Derivative(v(t), t) - 9.8, 0)

In [3]:
classify_ode(ecuacion_diferencial)

('factorable',
 'nth_algebraic',
 'separable',
 '1st_exact',
 '1st_linear',
 'Bernoulli',
 '1st_homogeneous_coeff_best',
 '1st_homogeneous_coeff_subs_indep_div_dep',
 '1st_homogeneous_coeff_subs_dep_div_indep',
 '1st_power_series',
 'lie_group',
 'nth_linear_constant_coeff_undetermined_coefficients',
 'nth_linear_euler_eq_nonhomogeneous_undetermined_coefficients',
 'nth_linear_constant_coeff_variation_of_parameters',
 'nth_linear_euler_eq_nonhomogeneous_variation_of_parameters',
 'nth_algebraic_Integral',
 'separable_Integral',
 '1st_exact_Integral',
 '1st_linear_Integral',
 'Bernoulli_Integral',
 '1st_homogeneous_coeff_subs_indep_div_dep_Integral',
 '1st_homogeneous_coeff_subs_dep_div_indep_Integral',
 'nth_linear_constant_coeff_variation_of_parameters_Integral',
 'nth_linear_euler_eq_nonhomogeneous_variation_of_parameters_Integral')

In [4]:
funcion_velocidad = dsolve(ecuacion_diferencial, hints = 'separable').subs(C1, 0)
funcion_velocidad

Eq(v(t), -9.8*t)

# Identificación de esta función

* la función velocidad es lineal de pendiente negativa 
* Como no tiene término independiente, entonces, su gráfico pasa por el origen

In [17]:
import pandas as pd 
import numpy as np 
tiempos = np.array([0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.64])

def v(t):
    return -9.8*t

velocidades = v(tiempos)
tabla_velocidades = pd.DataFrame({'tiempo (segundos)':tiempos, 'velocidades(m/s)':velocidades })
tabla_velocidades

tiempo (segundos)  velocidades(m/s)
0               0.00            -0.000
1               0.10            -0.980
2               0.20            -1.960
3               0.30            -2.940
4               0.40            -3.920
5               0.50            -4.900
6               0.60            -5.880
7               0.64            -6.272

* a medida que el tiempo pasa la velocidad decrece. 

In [24]:
from plotly import graph_objects as go 

fig = go.Figure()  

fig.add_trace(go.Scatter(x = tiempos, y = velocidades, mode = 'markers' ))

# Título del gráfico
fig.update_layout(title='Velocidad en función del Tiempo')

# Etiquetas para los ejes
fig.update_xaxes(title_text='Tiempo')
fig.update_yaxes(title_text='Velocidad')

# Mostrar el gráfico
fig.show()

## Descripción verbal a partir de la lectura de las representaciones algebraica, tabular y gráfica de la función de velocidad. 

1.  
2.  
3.  

In [5]:
ecuacion_dos = funcion_velocidad.subs(v, diff(h))

ecuacion_dos  # esto es gracias a Newton

Eq(Derivative(h(t), t), -9.8*t)

In [6]:
ecuacion_tres = dsolve(ecuacion_dos, hints = 'separable')

ecuacion_tres 

Eq(h(t), C1 - 4.9*t**2)

In [7]:
ecuacion_cuatro = dsolve(ecuacion_dos).subs(C1, 2)

ecuacion_cuatro 

Eq(h(t), 2 - 4.9*t**2)

In [ ]:
# tabla para la función altura
import pandas as pd 
import numpy as np 
tiempos = np.array([0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.64])

def h(t):
    return (-9.8/2)*t**2 + 2 # h(t)=2−4.9t2

alturas = h(tiempos)
tabla_alturas = pd.DataFrame({'tiempo (segundos)':tiempos, 'alturas(m)':alturas })
tabla_alturas

tiempo (segundos)  alturas(m)
0               0.00     2.00000
1               0.10     1.95100
2               0.20     1.80400
3               0.30     1.55900
4               0.40     1.21600
5               0.50     0.77500
6               0.60     0.23600
7               0.64    -0.00704

In [26]:
from plotly import graph_objects as go 

fig = go.Figure()  

fig.add_trace(go.Scatter(x = tiempos, y = alturas, mode = 'markers' ))

# Título del gráfico
fig.update_layout(title='Altura en función del Tiempo')

# Etiquetas para los ejes
fig.update_xaxes(title_text='Tiempo')
fig.update_yaxes(title_text='Altura')

# Mostrar el gráfico
fig.show()

## Descripción verbal a partir de la lectura de las representaciones algebraica, tabular y gráfica de la función de altura $h = h(t)$. 

1.  Descripción a partir de la representación algebraica: 
2.  Descripción a partir de la representación tabular:
3.  Descripción a partir de la representación gráfica: 

In [10]:
ecuacion_cinco = ecuacion_cuatro.subs(h, 0)
ecuacion_cinco 

Eq(0, 2 - 4.9*t**2)

In [14]:
tiempo_que_tarda_en_caer = round(solve(ecuacion_cinco)[1], 2)
tiempo_que_tarda_en_caer

0.64

In [13]:
funcion_velocidad.subs(t, tiempo_que_tarda_en_caer)

Eq(v(0.63887656499994), -6.26099033699941)

## Conclusión verbal o respuesta a las preguntas planteadas en el problema

El tiempo que tarda en caer el marcador al suelo es de 0.64 segundos y la velocidad con la que llga al suelo es de -6.26 metros por segundo.  

  
# Ejemplo de parcial   
## Situación problemática planteada por el profesor Marco Cañas:

El profesor deja caer un marcador de tablero desde una altura de 2 metros. 
Determine: 

1. Las funciones de velocidad y altura que le permitiran describir el movimiento y predecir mucha información sobre este interesante fenómeno.  


## Dificultades de aprendizaje detectados durante la clase

1. Reconocer el modelo o ecuación diferencial, $\frac{dv}{dt} = \text{constante}$, que representa el fenómeno de caída libre cuando se desprecia el rozamiento con el aire
2. Resolver problemas sobre caída libre partiendo del modelo o ecuación diferencial que representa este fenónemo: $\frac{dv}{dt} = \text{constante}$. 
3. Los estudiantes no están razonando haciendo uso del concepto de función $v = v(t)$ y de $h = h(t)$
4. No están aplicando el razonamiento de integral para obtener la función de su derivada:

$$\frac{dv}{dt}  \quad \quad \overbrace{\quad \Rightarrow \quad }^{ \int } \quad \quad v(t) + C_{1} $$

In [1]:
# Programa para valoración cuantitativa por meses para los estudiantes de todos los cursos
import numpy as np 
import pandas as pd   

# Weights
weightAssistance = 12/100
weightAttentionParticipation = 8/100
weightHandwrittenNotebook = 8/100
weightJupyterNotebook = 20/100
weightQuiz = 16/100
weightPartial = 48/100

url_group = 'https://raw.githubusercontent.com/marco-canas/groups_list/main/seminario_didactica_aritmetica.csv'

df = pd.read_csv(url_group, decimal = ',') 

df.Assistance = (df.Assistance/np.max(df.Assistance))*5.0
df.AttentionParticipation = (df.AttentionParticipation/np.max(df.AttentionParticipation))*5.0
df.HandwrittenNotebook = (df.HandwrittenNotebook/np.max(df.HandwrittenNotebook))*5.0
df.JupyterNotebook = (df.JupyterNotebook/np.max(df.JupyterNotebook))*5.0  


df['FollowUp'] = np.round(df.Assistance*weightAssistance+\
    df.AttentionParticipation*weightAttentionParticipation+\
    df.HandwrittenNotebook*weightHandwrittenNotebook+\
    df.JupyterNotebook*weightJupyterNotebook+\
    df.Quiz*weightQuiz+\
    df.Partial*weightPartial,1)
df[['Nombre', 'FollowUp']] 

Nombre  FollowUp
0                 ARIAS ARRIETA FREDY       NaN
1       BOHORQUEZ NAVARRO DINA ESTELA       NaN
2        CARDONA RAMIREZ SARA PAULINA       NaN
3             CASTANEDA REINOZA DIEGO       NaN
4             ESPEJO PEREZ LUIS DAVID       NaN
5   HERNANDEZ GUERRA FABIAN ALEJANDRO       NaN
6        LUCAS FLOREZ NAYELIS NATALIA       NaN
7    MARTINEZ BALLESTEROS VICTOR HUGO       NaN
8        MARTINEZ MENDOZA JAIR MIGUEL       NaN
9           OCHOA ARRIETA JUAN ANDRES       NaN
10      RODRIGUEZ VERONA CARLOS MARIO       NaN
11           RUIZ CARDENAS LUIS MARIO       NaN
12         SIERRA OLIVEROS OMAR DARIO       NaN
13     SUAREZ RESTREPO MARIA ANGELICA       NaN
14             TAMAYO DE LA OSSA SARA       NaN

## Estrategias propuestas para la solución de las dificultades de aprendizaje detectadas

## Grabaciones en audio para fomentar la capacidad de memorizar enunciados de teoremas y resultados científicos y la capacidad de parafrasear estos resultados. 

Se dispondrá de un lugar en el drive asociuado al correo institucional de la Universidad Nacional del Profesor Marso para albergar los archivosd de audio de fomento a la memoirización, enunciación y parafraseo de los resultados científicos, prtopiedades, reglas, etc. 

* Vocaroo

# Vocabulary 

# FODA del grupo y del profesor  

* F: Fortalezas
* O: Oportunidades
* D: Dificultades
* A: Amenazas.

## ¿Cómo se sintió en este espacio de enseñanza y aprendizaje?

!Dios te bendiga¡

## Referentes 

* [Documentación de graficación con plotly.express y plotly. graph_objets](https://plotly.com/python/creating-and-updating-figures/)
* [Documentación de la función dsolve](https://docs.sympy.org/latest/modules/solvers/ode.html)

* [stewart precálculo](https://udeaeduco-my.sharepoint.com/:b:/g/personal/marco_canas_udea_edu_co/ERHIq62I6qFNrmxy6LZb8ZMBci7kUsyNME1nIh9yCBMJ_w?e=0zMSla)  

* [stewart cálculo](https://udeaeduco-my.sharepoint.com/:b:/g/personal/marco_canas_udea_edu_co/EZgXZjAp8QxPqOAim2hs6LcBNPLGjSHf-xwYnUVYkwa04w?e=RZdTCy)  

* [larson](https://udeaeduco-my.sharepoint.com/:b:/g/personal/marco_canas_udea_edu_co/ES71ChFeO9ZDhW3TwC5Ijk8BjxUK3Pdqz_fjHxTTFAfIAg?e=VDEjfu)

* [uzcategui](https://udeaeduco-my.sharepoint.com/:b:/g/personal/marco_canas_udea_edu_co/ETDikm-lVl1Or8XoEo9oyh0BEti9Zs8le-f0D-dBdtZmbA?e=bBsoyQ)

* [Recomendación de la UNESCO sobre ciencia abierta](https://unesdoc.unesco.org/ark:/48223/pf0000379949_spa)  

* [chatGPT](https://openai.com/blog/chatgpt)  

* Géron, A. (2017). Hands-on machine learning with scikit-learn and tensorflow: Concepts. Tools, and Techniques to build intelligent systems.[Geron, Aurelien. Hands on Machine Learning](https://udeaeduco-my.sharepoint.com/:b:/g/personal/marco_canas_udea_edu_co/Ecet27yjQzZIlT1Y_Bc2erkBUhbeYuIe6HG8i1FYZRlZww?e=dlvZ4v)  

* McKinney, W. (2012). Python for data analysis: Data wrangling with Pandas, NumPy, and IPython. " O'Reilly Media, Inc.".[Wes Mackinney. ](https://udeaeduco-my.sharepoint.com/:b:/g/personal/marco_canas_udea_edu_co/EVbi5JIeBl9ErbiUnZfGe8YBhNTnZ8sxTK5hjIOPK4UpGw?e=fFclah)

### [Evaluamos al profesor Marco Cañas Aquí](https://forms.office.com/Pages/ResponsePage.aspx?id=IefhmYRxjkmK_7KtTlPBwkanXIs1i1FEujpsZgO6dXpUREJPV1kxUk1JV1ozTFJIQVNIQjY5WEY3US4u)

## [Evaluación luego de alcanzar estos objetivos de aprendizaje]()

### Continue su aprendizaje en la siguiente clase a través del siguiente [vínculo]()

# Conjeturas pedagógicas fruto de la aplicación del modelo de aprendizaje invertido y del enfoque hacia la ciencia de datos con python

1. Todo cálculo o resultado debe ser interpretado en una línea markdown del cuaderno Jupyter, inmediatamente después de la enunciación del resultado y después de la presentación de una tabla o gráfico bidimensional, de tal menera que el estudiante explicite la comprensión verbal del resultado y las inferencias o estrategias que este resultado le sugieren.   